10-std.tex

```diff
@@ -368,6 +368,98 @@ \section{Lambda}
+	\item[\ic{::expr.field::}] field access
+	\item[\ic{::(expr)::}] the expression expr is evaluated
+	\item[\ic{::(e1 op e2)::}] the operation op is applied to e1 and e2
+	\item[\ic{::(135)::}] the integer 135. Float constants are not allowed
+\end{description}
+
+\paragraph{Conditions}
+It is possible to test conditions using \ic{::if flag1::}. Optionally, the condition may be followed by \ic{::elseif flag2::} or \ic{::else::}. Close the condition with \ic{::end::}.
+
+\begin{lstlisting} 
+::if isValid:: valid ::else:: invalid ::end::
+\end{lstlisting} 
+
+Operators can be used, but it doesn't deal with operator precedence. Therefore it is required to enclose each operation in parentheses \ic{()}. The current operators are allowed: \ic{+}, \ic{-}, \ic{*}, \ic{/}, \ic{>}, \ic{<},  \ic{>=}, \ic{<=}, \ic{==}, \ic{!=}, \ic{&&} and \ic{||}.
+
+For example \ic{::((1 + 3) == (2 + 2))::} will display true. 
+
+\begin{lstlisting} 
+::if (points == 10):: Great! ::end::
+\end{lstlisting} 
+
+To compare to a string, use double quotes \ic{"} in the template.
+\begin{lstlisting} 
+::if (name == "Mark"):: Hi Mark ::end::
+\end{lstlisting} 
+
+\paragraph{Iterating}
+Iterate on a structure by using \ic{::foreach::}. End the loop with \ic{::end::}.
+\begin{lstlisting} 
+<table>
+	<tr>
+		<th>Name</th>
+		<th>Age</th>
+	</tr>
+	::foreach users::
+		<tr>
+			<td>::name::</td>
+			<td>::age::</td>
+		</tr>
+	::end::
+</table>
+\end{lstlisting} 
+
+\paragraph{Sub-templates}
+To include templates inside other templates, pass the sub-template result string as a parameter.
+\begin{lstlisting} 
+var users = [{name:"Mark", age:30}, {name:"John", age:45}];
+
+var userTemplate = new haxe.Template("::foreach users:: ::name::(::age::) ::end::");
+var userOutput = userTemplate.execute({users: users});
+
+var template = new haxe.Template("The users are ::users::");
+var output = template.execute({users: userOutput});
+trace(output);
+\end{lstlisting} 
+The console will trace \ic{The users are Mark(30) John(45)}.
+
+\paragraph{Macros}
+To callback code while parts of the template are rendered, providing a \expr{macros} object to the argument of \expr{Template.execute}. The key will act the template variable name, the value refers to a callback function that should return a \type{String}. The first argument of this macro function is always a \expr{resolve()} method, followed by the given arguments. The resolve function can be called to retreive values from the Template context. If \expr{macros} has no such field, the result is unspecified.
+
+The following example passes itself as macro function context, and executes \ic{display} from the template.
+\haxe{assets/TemplateMacros.hx}
+The console will trace \ic{The results: Mark ran 3.5 kilometers in 15 minutes}.
+
+\paragraph{Globals}
+Use the \expr{Template.globals} object to store those values that should be applied across all Template instances. This has lower priority than the context argument of \expr{Template.execute}.
```

Comment:
"the values" instead of "those" (those implies distance). "template instances" (no caps)